kpops/component_handlers/topic/handler.py

```diff
@@ -127,14 +97,46 @@ async def __dry_run_topic_creation(
         except TopicNotFoundException:
             log.info(
                 greenify(
-                    f"Topic Creation: {topic_name} does not exist in the cluster. Creating topic."
+                    f"Topic Creation: {topic.name} does not exist in the cluster. Creating topic."
                 )
             )
             log.debug(f"POST /clusters/{self.proxy_wrapper.cluster_id}/topics HTTP/1.1")
             log.debug(f"Host: {self.proxy_wrapper.url}")
             log.debug(HEADERS)
             log.debug(topic_spec.model_dump())
 
+    async def __execute_topic_creation(
+        self, topic: KafkaTopic, topic_spec: TopicSpec
+    ) -> None:
+        try:
+            await self.proxy_wrapper.get_topic(topic.name)
+            topic_config_in_cluster = await self.proxy_wrapper.get_topic_config(
+                topic.name
+            )
+            differences = self.__get_topic_config_diff(
+                topic_config_in_cluster, topic.config.configs
+            )
+
+            if differences:
+                json_body = []
+                for difference in differences:
+                    if difference.diff_type is DiffType.REMOVE:
+                        json_body.append(
+                            {"name": difference.key, "operation": "DELETE"}
+                        )
+                    elif config_value := difference.change.new_value:
+                        json_body.append(
+                            {"name": difference.key, "value": config_value}
+                        )
+                await self.proxy_wrapper.batch_alter_topic_config(topic.name, json_body)
+
+            else:
+                log.info(
+                    f"Topic Creation: config of topic {topic.name} didn't change. Skipping update."
+                )
+        except TopicNotFoundException:
```

Comment:
Not part of this PR but maybe we should look at the proxy wrapper and improve this logic. Back then we handled these with exception but maybe this would be cleaner (also more performant) with an if-else check.

Reply:
I called it `validate_topics` based on the usage of a Pydantic "`BeforeValidator`" and the examples in their docs. Essentially it is a custom deserializer (that's what the "before" indicates). 
from their docs:
> Before validators run before Pydantic's internal parsing and validation (e.g. coercion of a str to an int). These are more flexible than After validators since they can modify the raw input, but they also have to deal with the raw input, which in theory could be any arbitrary object.

The actual validation is done during by Pydantic during the init. If you want I can name it `deserialize_topics` instead

Reply:
what do you have in mind to improve the performance here?
I agree that this (and most of our other handlers and API wrappers) needs refactoring, but I believe the exception is a good pattern in this case.

Reply:
I got mixed up with an article I read recently for Java. In Java this is considered as an anti-pattern. But apparently it's okay in Python. Here a short explanation:

https://stackoverflow.com/a/1835844